File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        queue<pair<TreeNode*, int>> q;
        if(!root)
            return result;
        q.push(pair<TreeNode*, int>(root, 0));
        while(q.size()) {
           
            TreeNode* t = q.front().first;
            int level = q.front().second;
            if(level < result.size())
                result[level].push_back(t->val);
            else{
                result.push_back({t->val});
            }
            q.pop();
            if(t->left)
                q.push(pair<TreeNode*, int>(t->left, level + 1));
            if(t->right)
                q.push(pair<TreeNode*, int>(t->right, level + 1));
            
        }
        
        return result;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        l = []
        ans = []
        l.append(root)
        level_count = 1
        while len(l) > 0:
            row = []
            count = level_count
            level_count = 0
            while count:
                t = l[0]
                l.pop(0)
                if t.left:
                    l.append(t.left)
                    level_count += 1
                if t.right:
                    l.append(t.right)
                    level_count += 1
                count -= 1
                row.append(t.val)
            ans.append(row)
        return ans

                





        